src/commands/flinkStatements.ts

```diff
@@ -86,6 +88,29 @@ export async function viewStatementSqlCommand(statement: FlinkStatement): Promis
   vscode.languages.setTextDocumentLanguage(doc, "flinksql");
   await vscode.window.showTextDocument(doc, { preview: false });
 }
+/**
+ * Monitors a Flink statement and fires the UDF change emitter when a CREATE_FUNCTION statement completes.
+ *
+ * @param statement - The FlinkStatement to monitor for CREATE_FUNCTION completion
+ * @param database - The CCloudFlinkDbKafkaCluster where the UDF will be created and which should be notified of changes
+ * @returns Promise that resolves when monitoring is complete (immediately if not a CREATE_FUNCTION, or after completion if it is)
+ */
+export async function fireEmitterWhenFlinkStatementIsCreatingFunction(
+  statement: FlinkStatement,
+  database: CCloudFlinkDbKafkaCluster,
+): Promise<void> {
+  if (statement?.status.traits?.sql_kind !== "CREATE_FUNCTION") {
+    return;
+  }
+
+  const completedStatement = await waitForStatementCompletion(statement);
+
+  if (completedStatement.status.phase !== Phase.COMPLETED) {
+    return;
+  }
+
+  udfsChanged.fire(database);
```

Comment:
If you want to clicktest _really_ deeply, set a breakpoint here and make sure it doesn't get hit when you run a `SELECT *` statement somewhere